File: app/src/main/java/com/falcon/restaurants/screens/splash/SplashViewModelFactory.java
package com.falcon.restaurants.screens.splash;

import android.app.Application;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import com.falcon.restaurants.screens.restaurant.RestaurantViewModel;
import com.falcon.restaurants.screens.meal.MealViewModel;

public class SplashViewModelFactory implements ViewModelProvider.Factory {

    Application application ;
    RestaurantViewModel restaurantViewModel;
    MealViewModel mealViewModel;

    public SplashViewModelFactory(Application application,
                                  RestaurantViewModel restaurantViewModel,
                                  MealViewModel mealViewModel) {
        this.application = application;
        this.restaurantViewModel = restaurantViewModel;
        this.mealViewModel = mealViewModel;
    }

    @androidx.annotation.NonNull
    @Override
    public <T extends ViewModel> T create(@androidx.annotation.NonNull Class<T> modelClass) {
        return (T) new SplashViewModel(application, restaurantViewModel, mealViewModel);
    }
}

File: app/src/main/java/com/falcon/restaurants/screens/splash/SplashViewModelFactory.kt
package com.falcon.restaurants.screens.splash

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.falcon.restaurants.screens.restaurant.RestaurantViewModel
import com.falcon.restaurants.screens.meal.MealViewModel

class SplashViewModelFactory(
    var application: Application,
    var restaurantViewModel: RestaurantViewModel,
    var mealViewModel: MealViewModel
): ViewModelProvider.Factory {

    override fun <T :ViewModel> create(modelClass:  Class<T>) : T {
        return SplashViewModel(application, restaurantViewModel, mealViewModel) as T
    }
}